### PR TITLE
Put a detailed comment in the apidocs

### PR DESCRIPTION
**Pull request by jessitron**
this diagram was useful to Lieven and me today, as he's making an inspection for the blog.
I want to include it in the API docs.

However, there's an open question in there for @johnsonr that really needs answered.

### CODE REVIEW

**Comment by cdupuis** [comment]
I removed it from the docs as it looks complete broken in the TypeDocs. We have to fix it first.

File: lib/api-helper/listener/executeAutoInspects.ts [hunk 1]
```diff
@@ -109,15 +111,12 @@ export function executeAutoInspects(autoInspectRegistrations: Array<AutoInspectR
...
-
-// tslint:enable
-/**
- * Apply code inspections
+ *
  * @param goalInvocation
  * @param autoInspectRegistrations
  * @param reviewListeners
  */
+// tslint:enable
```
Comment (cdupuis) [inline lib/api-helper/listener/executeAutoInspects.ts]: This won't show nicely in TypeDoc

screenshot 2018-10-19 at 17 35 59

**Comment by ddgenome** [comment]
https://stackoverflow.com/questions/2961310/jsdoc-adding-real-code-in-documentation

File: lib/api-helper/listener/executeAutoInspects.ts [hunk 2]
```diff
@@ -73,11 +73,11 @@ export function executeAutoInspects(autoInspectRegistrations: Array<AutoInspectR
 
 // tslint:disable
 /**
- * each inspection can return a result, which may be turned into a PushImpactResponse by its onInspectionResult,
+ * Each inspection can return a result, which may be turned into a PushImpactResponse by its onInspectionResult,
  * OR it may return a ProjectReview, which will be processed by each ProjectReviewListener. The Listener may also return a PushImpactResponse.
  * Each of these PushReactionResponses may instruct the AutoInspect goal to fail or to require approval.
  *
- * ROD: which of these paths is deprecated? I'm guessing the ReviewListener is deprecated
```
Comment (johnsonr) [inline lib/api-helper/listener/executeAutoInspects.ts]: The ReviewListener isn't deprecated... I don't think anything is deprecated here

**Comment by johnsonr** [comment]
What's the status of this?

**Pull request closed without merging**